File: add_account.rb
require 'yaml'
require_relative './core/o_auth'
require_relative './core/configger'

setting = YAML.load_file(File.expand_path('../settings.yml', __FILE__))
auth = O_auth.new(setting['consumer_key'], setting['consumer_secret'])

url = auth.get_url
system("start #{url}")
print('Enter pin code :')
pin = gets
token, secret = auth.get_token(pin.to_i)

client = Twitter::REST::Client.new do |conf|
  conf.consumer_key = setting['consumer_key']
  conf.consumer_secret = setting['consumer_secret']
  conf.access_token = token
  conf.access_token_secret = secret
end
uid = client.user.id

config = {
    'uid': uid,
    'consumer_key': setting['consumer_key'],
    'consumer_secret': setting['consumer_secret'],
    'access_token': token,
    'access_token_secret': secret
}

YAML.dump(config, open(File.expand_path("../data/conf/#{uid}.conf", __FILE__)))

File: add_account.py
# -- coding: utf-8 --
import tweepy
import json
import webbrowser
from core.client import Client
from core.setting import Setting

if __name__ == '__main__':
    setting = Setting()
    auth = tweepy.OAuthHandler(setting['key']['consumer_key'], setting['key']['consumer_secret'])
    url = auth.get_authorization_url()
    webbrowser.open_new(url)
    print(f'if don`t open Twitter authorization, please access {url} and accept.')
    print('Enter pin code : ', end='')
    pin = input()
    auth.get_access_token(pin)
    client = Client(
        consumer_key=setting['consumer_key'],
        consumer_secret=setting['consumer_secret'],
        access_token=auth.access_token,
        access_token_secret=auth.access_token_secret
    )
    uid = client.api.me().id

    config = {
        'user': {
            'id': int(uid),
            'screen_name': client.api.me().screen_name,
            'consumer_key': setting['consumer_key'],
            'consumer_secret': setting['consumer_secret'],
            'access_token': auth.access_token,
            'access_token_secret': auth.access_token_secret
        },
        'following': client.get_following(),
        'follower': client.get_follower()
    }

    f = open(f'./data/{uid}.json', 'w')
    json.dump(config, f, indent=True)
    f.close()

    print('Account successfully added!')
    input('Press enter to close.')
